sklearn/ensemble/forest.py

```diff
@@ -242,6 +243,10 @@ def fit(self, X, y, sample_weight=None):
         -------
         self : object
         """
+        # Validate hyperparameters
+        if self.n_estimators == 10:
```

Comment:
Often we would do something like change the default to None, and treat it as "warn and use 10" to avoid warning when the user explicitly sets n_estimators=10

Reply:
Is this in the dev guide? I don't think so, right? Also that deprecations should be in ``fit`` is not, I think? New issue? ;)

Reply:
@jnorthman, I'm not sure I understand you. 
Are you advising I change the `n_estimators` value to `None`?

Reply:
@amueller I've created two issues to take care of both of these.

Reply:
Yes, defaulting n_estimators=None would be fine. n_estimators='warn'

Reply:
Okay.
But I'm a bit confused now because @amueller suggested the default for n_estimators should still be 10
[here](https://github.com/scikit-learn/scikit-learn/pull/11172/files/8b5cb95c87b22131ed0f55205717accd4893f5bf#r194237352).

Am I missing something?